test/dual_reduce/dual_reduce.cpp

```diff
@@ -0,0 +1,178 @@
+#include "getopt.h"
+
+#include "host_common_util.hpp"
+#include "profile_dual_reduce_impl.hpp"
+
+using namespace ck;
+
+static struct option long_options[] = {{"inLengths", required_argument, nullptr, 'D'},
+                                       {"reduceDimensions", required_argument, nullptr, 'R'},
+                                       {"scales", required_argument, nullptr, 'S'},
+                                       {"help", no_argument, nullptr, '?'},
+                                       {nullptr, 0, nullptr, 0}};
+
+class SimpleAppArgs
```

Comment:
The test cases only differ in reduce dimensions. Can you test other input argument combos as well? 

Also, the test file should be as stripped down as possible so I imagine that `SimpleAppArgs` and other things unrelated to test itself should be removed.

Reply:
> The test cases only differ in reduce dimensions. Can you test other input argument combos as well?

Good suggestion !

> 
> Also, the test file should be as stripped down as possible so I imagine that `SimpleAppArgs` and other things unrelated to test itself should be removed.

If want to support various tensor sizes for testing,  need `SimpleAppArgs` to manage the input parameters.